crates/ruff/src/settings/options_base.rs

```diff
@@ -1,42 +1,180 @@
-pub trait ConfigurationOptions {
-    fn get_available_options() -> Vec<(&'static str, OptionEntry)>;
+use std::fmt::{Display, Formatter};
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum OptionEntry {
+    Field(OptionField),
+    Group(OptionGroup),
+}
+
+impl Display for OptionEntry {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            OptionEntry::Field(field) => field.fmt(f),
+            OptionEntry::Group(group) => group.fmt(f),
+        }
+    }
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct OptionGroup(&'static [(&'static str, OptionEntry)]);
+
+impl OptionGroup {
+    pub const fn new(options: &'static [(&'static str, OptionEntry)]) -> Self {
+        Self(options)
+    }
+
+    pub fn iter(&self) -> std::slice::Iter<(&str, OptionEntry)> {
+        self.into_iter()
+    }
 
     /// Get an option entry by its fully-qualified name
     /// (e.g. `foo.bar` refers to the `bar` option in the `foo` group).
-    fn get(name: Option<&str>) -> Option<OptionEntry> {
-        let mut entries = Self::get_available_options();
+    ///
+    /// ## Examples
+    ///
+    /// ### Find a direct child
+    ///
+    /// ```rust
+    /// # use ruff::settings::options_base::{OptionGroup, OptionEntry, OptionField};
```

Comment:
Maybe a ridiculous question, but is there any way to get Rust to compile these snippets, to ensure they're kept up to date?

Reply:
Even better. The snippets run as part of our test suite ([doctests](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html)) This is one of my favorite parts of Rust: writing documentation with examples gives you testing for free ☺️

Reply:
Ohh I assumed something like this was happening, but `cargo check --all` didn't have any effect for me after introducing errors.